x-pack/plugins/ml/public/explorer/explorer_controller.js

```diff
@@ -439,6 +424,70 @@ module.controller('MlExplorerController', function (
     });
   };
 
+  function loadAnomalies(jobIds, influencers, earliestMs, latestMs) {
+    // Loads the anomalies for the table, plus the scores for
+    // the Top Influencers List for the influencers in the anomaly records.
+
+    if (influencers.length === 0) {
+      getTopInfluencers(jobIds, earliestMs, latestMs);
+    }
+
+    mlResultsService.getRecordsForInfluencer(
+      jobIds, influencers, 0, earliestMs, latestMs, 500
+    )
+      .then((resp) => {
+        if (influencers.length > 0) {
+          // Filter the Top Influencers list to show just the influencers from
+          // the records in the selected time range.
+          const recordInfluencersByName = {};
+          resp.records.forEach((record) => {
+            const influencersByName = record.influencers || [];
+            influencersByName.forEach((influencer) => {
+              const fieldName = influencer.influencer_field_name;
+              const fieldValues = influencer.influencer_field_values;
+              if (recordInfluencersByName[fieldName] === undefined) {
+                recordInfluencersByName[fieldName] = [];
+              }
+              recordInfluencersByName[fieldName].push(...fieldValues);
+            });
+          });
+
+          const uniqValuesByName = {};
+          Object.keys(recordInfluencersByName).forEach((fieldName) => {
+            const fieldValues = recordInfluencersByName[fieldName];
+            uniqValuesByName[fieldName] = _.uniq(fieldValues);
+          });
+
+          const filterInfluencers = [];
+          Object.keys(uniqValuesByName).forEach((fieldName) => {
+            // Find record influencers with the same field name as the clicked on cell(s).
+            const matchingFieldName = influencers.filter((influencer) => {
+              return influencer.fieldName === fieldName;
+            });
+
+            if (matchingFieldName.length > 0) {
+              // Filter for the value(s) of the clicked on cell(s).
+              filterInfluencers.push(...influencers);
+            } else {
+              // For other field names, add values from all records.
+              uniqValuesByName[fieldName].forEach((fieldValue) => {
+                filterInfluencers.push({ fieldName, fieldValue });
+              });
+            }
+          });
+
+          getTopInfluencers(jobIds, earliestMs, latestMs, filterInfluencers);
+        }
+
+        // Need to use $timeout to ensure the update happens after the child scope is updated with the new data.
```

Comment:
Comment talks about `$timeout` but `$evalAsync` is used in the code.

Reply:
`$evalAsync` is probably the better choice of the two